presto-main/src/main/java/io/prestosql/operator/TaskOutputOperator.java

```diff
@@ -122,8 +123,13 @@ public boolean isFinished()
     @Override
     public ListenableFuture<?> isBlocked()
     {
-        ListenableFuture<?> blocked = outputBuffer.isFull();
-        return blocked.isDone() ? NOT_BLOCKED : blocked;
+        if (isBlocked.isDone()) {
+            isBlocked = outputBuffer.isFull();
```

Comment:
future of output buffer is already cached

Reply:
Same justification as [my reply](https://github.com/prestosql/presto/pull/6260#discussion_r540921443) about `PartitionedOutputOperator`